wpilibc/src/main/native/include/frc/buttons/POVButton.h

```diff
@@ -0,0 +1,29 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include "frc/GenericHID.h"
+#include "frc/buttons/Button.h"
+
+namespace frc {
+class POVButton : public Button {
+
+ public:
+  POVButton(GenericHID* joystick, int angle, int povNumber);
```

Comment:
Use references instead of pointers in the constructor.

Reply:
Those classes are legacy and don't do things idiomatically. In a nutshell, value semantics are preferred in user-facing APIs over reference semantics in modern C++ types (yes, references still have reference semantics, but syntactically they look just like values). Every new API we've added since 2017 has used references in the constructor. It would be worth it (in a separate PR) to add reference overloads to the other classes in addition to the pointer constructors (we can't remove the pointer versions for backwards compat). We did this for PIDController a while ago, for example, and it makes the API easier to use because users don't need to mess with random ampersands.